spec.html

```diff
@@ -547,7 +547,10 @@ <h1>InnerModuleEvaluation ( _module_, _stack_, _index_ )</h1>
 
       <emu-alg>
         1. If _module_ is not a Cyclic Module Record, then
-          1. Perform ? _module_.Evaluate().
+          1. Let _promise_ be ! _module_.Evaluate().
+          1. Assert: _promise_.[[PromiseState]] is not ~pending~.
```

Comment:
Surely we should permit pending promises for eg Web Assembly support?

Reply:
My understanding is that only Cyclic modules can be async. If not, I believe we need to move a whole lot of state from Cyclic to Abstract.

Reply:
> My understanding is that only Cyclic modules can be async.

What constraint makes you think this?

Ah, yes sorry of course Web Assembly modules are implemented as Cyclic module records yes. That said, it might still be useful for other non-cyclic modules that are clear graph exit points like CSS or synthetic modules right?

Reply:
Possibly - I'd like to postpone doing that work until there's a clearer need, though.